csharp/CMakeLists.txt

```diff
@@ -265,18 +235,36 @@ add_custom_target(csharp_sdk ALL
 )
 
 if( BUILD_NUGET_PACKAGE )
+  if(NOT CMAKE_CL_64)
+    message(FATAL_ERROR "Nuget package cannot be built using the 64 bit generator")
+  endif()
```

Comment:
Hum. I'm confused by this. Are we sure no-one wants a 64bit nuget? As I understand, there's a way to make multi-config nugets too (eg you'd have a directory structure per platform / arch and it would select the right one).

Reply:
If you are doing an entirely new build tree, and specifiying to use MSVC -A Win32, do we care if the root call was with 64?

Reply:
I think the words in the error message are a mistake. It should say "Nuget package cannot be built using the 32 bit generator." That is not to say that you can't have 32 bit binaries in the nuget package. To the contrary that is exactly what this PR is all about, which is building a universal 32/64 bit package. However. to make that you need to configure build_nuget_package on from a 64 bit generator and then the 32 bit build will be carried out as an external project and then included into the nuget package.

Reply:
I will update the error message.

Reply:
Oh ok got it, yeah with the updated error message to read "Nuget package can only be built using the 64 bit generator" I'm good with it!